apps/sms/test/marionette/lib/messages.js

```diff
@@ -208,27 +208,21 @@ var ConversationAccessor = require('./views/conversation/accessors');
         },
 
         performHeaderAction: function() {
-          this.Composer.header.scriptWith(function(header) {
-            var event = document.createEvent('HTMLEvents');
-            event.initEvent('action', true, true);
-            header.dispatchEvent(event);
-          });
+          this.client.switchToShadowRoot(this.Composer.header);
+          this.Composer.headerActionButton.tap();
+          this.client.switchToShadowRoot();
         },
 
         performReportHeaderAction: function() {
-          this.Report.header.scriptWith(function(header) {
-            var event = document.createEvent('HTMLEvents');
-            event.initEvent('action', true, true);
-            header.dispatchEvent(event);
-          });
+          this.client.switchToShadowRoot(this.Composer.header);
+          this.Report.headerActionButton.tap();
```

Comment:
So again, you use ```this.Report.headerActionButton```, but you didn't define it anywhere.

Please, let me know if any difficulties in finding where ```this.Reports``` is defined (basically almost at the beginning of the file).